gse_get_client_auth_token: Improve error handling

Handles an error in gss_init_sec_context when KDC does not support the requested
encryption type. This error is needed for Chromad to present the user a
reasonable error message.

crbug.com/806910.


diff --git a/source3/librpc/crypto/gse.c b/source3/librpc/crypto/gse.c
index bcda8de..5cb6647 100644
--- a/source3/librpc/crypto/gse.c
+++ b/source3/librpc/crypto/gse.c
@@ -526,6 +526,9 @@ init_sec_context_done:
 			/* Garbage input, possibly from the auto-mech detection */
 			status = NT_STATUS_INVALID_PARAMETER;
 			goto done;
+		case (OM_uint32)KRB5KDC_ERR_ETYPE_NOSUPP:
+			status = NT_STATUS_KDC_UNKNOWN_ETYPE;
+			goto done;
 		default:
 			DBG_ERR("gss_init_sec_context failed with [%s](%u)\n",
 				gse_errstr(talloc_tos(), gss_maj, gss_min),
